src/masternode/payments.cpp

```diff
@@ -12,19 +12,116 @@
 #include <governance/governance.h>
 #include <key_io.h>
 #include <logging.h>
+#include <llmq/utils.h>
 #include <masternode/sync.h>
 #include <primitives/block.h>
 #include <script/standard.h>
 #include <tinyformat.h>
 #include <util/ranges.h>
 #include <util/system.h>
 #include <validation.h>
+#include <evo/creditpool.h>
 
 #include <string>
 
-CMasternodePayments mnpayments;
+/**
+*   GetMasternodeTxOuts
+*
+*   Get masternode payment tx outputs
+*/
+
+static bool GetBlockTxOuts(const int nBlockHeight, const CAmount blockReward, std::vector<CTxOut>& voutMasternodePaymentsRet)
+{
+    voutMasternodePaymentsRet.clear();
+
+    CAmount masternodeReward = GetMasternodePayment(nBlockHeight, blockReward, Params().GetConsensus().BRRHeight);
+
+    const CBlockIndex* pindex = WITH_LOCK(cs_main, return ::ChainActive()[nBlockHeight - 1]);
+    bool fMNRewardReallocated =  llmq::utils::IsMNRewardReallocationActive(pindex);
+
+    if (fMNRewardReallocated) {
+        const CAmount platformReward = masternodeReward * 0.375;
+        masternodeReward -= platformReward;
+
+        assert(MoneyRange(platformReward));
+        assert(MoneyRange(masternodeReward));
+
+        LogPrintf("CMasternodePayments::%s -- MN reward %lld reallocated to credit pool\n", __func__, platformReward);
+        voutMasternodePaymentsRet.emplace_back(platformReward, CScript() << OP_RETURN);
```

Comment:
NOTE: using `OP_RETURN` in coinbase will most likely break p2pool

Reply:
it looks like p2pool miners are no longer active though, check [blockchair](https://blockchair.com/dash/outputs?s=time(desc)&q=is_from_coinbase(true),is_spendable(false)#f=transaction_hash,time,is_from_coinbase,script_bin,is_spendable) and [chainz](https://chainz.cryptoid.info/dash/extraction.dws?30.htm)